src/main/java/az/edu/turing/booking/service/impl/UserServiceImpl.java

```
@@ -29,32 +35,56 @@ public class UserServiceImpl implements UserService {
     @Transactional
     @Override
     public UserDto create(UserCreateRequest request) {
+        checkUsernameAlreadyExists(request.getUsername());
+        UserEntity userEntity = userMapper.toEntity(request);
+
+        return userMapper.toDto(userRepository.save(userEntity));
+    }
 
-        if (userRepository.existsByUsername(request.getUsername())) {
-            throw new AlreadyExistsException("User already exists with username: " + request.getUsername());
+    @Transactional
+    @Override
+    public UserDto create(Long id, String password, AdminCreateRequest request) {
+        if (!isAdmin(id)) {
+            throw new AccessDeniedException("User is not an admin");
+        }
+        if (!checkPassword(id, password)) {
+            throw new AccessDeniedException("Password is incorrect");
```

you can throw this exception inside checkPassword method